checks.d/spark.py

```diff
@@ -198,6 +201,7 @@ def __init__(self, name, init_config, agentConfig, instances=None):
         AgentCheck.__init__(self, name, init_config, agentConfig, instances)
         self.previous_jobs = {}
         self.previous_stages = {}
+        self._spark_pre_20_mode = False
```

Comment:
I don't think we need this as an attribute, it will change from instance to instance and it's not too expensive to look up in the instance dict (something we'd have to do anyway if you made the `self._spark_pre_20_mode` into a dict). We can probably just lookup the value in `check()` for the relevant code paths.